Sources/SpeziFHIR/FHIRStore.swift

```diff
@@ -19,149 +17,156 @@ import Spezi
+    @MainActor public var medications: [FHIRResource] {
         access(keyPath: \.medications)
-        return lock.withLock {
-            _resources.filter { $0.category == .medication }
-        }
+        return _resources.filter { $0.category == .medication }
     }
-    
-    /// Observations.
-    public var observations: [FHIRResource] {
+
+    /// `FHIRResource`s with category `observation`.
+    @MainActor public var observations: [FHIRResource] {
         access(keyPath: \.observations)
-        return lock.withLock {
-            _resources.filter { $0.category == .observation }
-        }
+        return _resources.filter { $0.category == .observation }
     }
-    
-    /// Other resources that could not be classified on the other categories.
-    public var otherResources: [FHIRResource] {
-        access(keyPath: \.otherResources)
-        return lock.withLock {
-            _resources.filter { $0.category == .other }
-        }
-    }
-    
-    /// Procedures.
-    public var procedures: [FHIRResource] {
+
+    /// `FHIRResource`s with category `procedure`.
+    @MainActor public var procedures: [FHIRResource] {
         access(keyPath: \.procedures)
-        return lock.withLock {
-            _resources.filter { $0.category == .procedure }
-        }
+        return _resources.filter { $0.category == .procedure }
     }
-    
-    
-    public required init() {
-        self._resources = []
+
+    /// `FHIRResource`s with category `other`.
+    @MainActor public var otherResources: [FHIRResource] {
+        access(keyPath: \.otherResources)
+        return _resources.filter { $0.category == .other }
     }
-    
-    
-    /// Inserts a FHIR resource into the store.
+
+
+    /// Create an empty ``FHIRStore``.
+    public required init() {}
+
+
+    /// Inserts a FHIR resource into the ``FHIRStore``.
     ///
     /// - Parameter resource: The `FHIRResource` to be inserted.
+    @MainActor
     public func insert(resource: FHIRResource) {
-        withMutation(keyPath: resource.storeKeyPath) {
-            lock.withLock {
-                _resources.append(resource)
-            }
-        }
+        _$observationRegistrar.willSet(self, keyPath: resource.category.storeKeyPath)
+
+        _resources.append(resource)
+
+        _$observationRegistrar.didSet(self, keyPath: resource.category.storeKeyPath)
     }
-    
-    /// Removes a FHIR resource from the store.
+
+    /// Inserts a ``Collection`` of FHIR resources into the ``FHIRStore``.
     ///
-    /// - Parameter resource: The `FHIRResource` identifier to be inserted.
-    public func remove(resource resourceId: FHIRResource.ID) {
-        lock.withLock {
-            guard let resource = _resources.first(where: { $0.id == resourceId }) else {
-                return
+    /// - Parameter resources: The `FHIRResource`s to be inserted.
+    public func insert<T: Collection>(resources: sending T) async where T.Element == FHIRResource {
```

Comment:
Should we make this method just be isolated to MainActor (and sync) instead of requiring that the input must be supplied sending? Seems a bit overly restrictive?

Reply:
Probably you're right, doesn't really make sense to make this things `async` (and non-isolated to the main actor), even though the function body mostly runs on the main actor. Just wanted to avoid performing the bulk insert of FHIR resources (probably as startup of app) completely on the main actor, but yeah, so be it.